File: backend/src/main/java/org/akashihi/mdg/dao/BudgetEntryModeConverter.java
package org.akashihi.mdg.dao;

import org.akashihi.mdg.entity.BudgetEntryMode;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

@Converter
public class BudgetEntryModeConverter implements AttributeConverter<BudgetEntryMode, String> {
    @Override
    public String convertToDatabaseColumn(BudgetEntryMode budgetEntryMode) {
        return budgetEntryMode.toDbValue();
    }

    @Override
    public BudgetEntryMode convertToEntityAttribute(String s) {
        return BudgetEntryMode.from(s);
    }
}

File: backend/src/main/kotlin/org/akashihi/mdg/dao/BudgetEntryModeConverter.kt
package org.akashihi.mdg.dao

import org.akashihi.mdg.entity.BudgetEntryMode
import javax.persistence.AttributeConverter
import javax.persistence.Converter

@Converter
class BudgetEntryModeConverter : AttributeConverter<BudgetEntryMode, String> {
    override fun convertToDatabaseColumn(budgetEntryMode: BudgetEntryMode): String {
        return budgetEntryMode.toDbValue()
    }

    override fun convertToEntityAttribute(s: String): BudgetEntryMode {
        return BudgetEntryMode.from(s)
    }
}
